File: java/benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/InitializationPhase.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.result.Status;

class InitializationPhase extends BenchmarkPhase {

    InitializationPhase(Connection connection) {
        super(connection);
    }

    @Override
    protected Status execute() {
        return getConnection().initialize();
    }

}

File: cpp/src/observatory/phase/InitializationPhase.cpp
#include "InitializationPhase.h"
#include <observatory/Benchmark.h>

#include <utility>

namespace Observatory {

InitializationPhase::InitializationPhase(Benchmark &benchmark) :
        BenchmarkPhase(benchmark) {}

const char* InitializationPhase::getName() {
    return "InitializationPhase";
}

Status InitializationPhase::execute() {
    return getBenchmark().initialize();
}

}